src/boot/efi/bcd.c

```diff
@@ -0,0 +1,286 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <efi.h>
+#include "macro-fundamental.h"
+#include "util.h"
+
+enum {
+        SIG_BASE_BLOCK = 1718052210, /* regf */
+        SIG_KEY = 27502,             /* nk */
+        SIG_SUBKEY_FAST = 26220,     /* lf */
+        SIG_KEY_VALUE = 27510,       /* vk */
+};
+
+enum {
+        REG_SZ = 1,
+        REG_MULTI_SZ = 7,
+};
+
+typedef struct {
+        UINT32 sig;
+        UINT32 primary_seqnum;
+        UINT32 secondary_seqnum;
+        UINT64 _pad1;
```

Comment:
Hmm, uint64_t would be aligned to 64 bits. Here `_packed_` is used, so the alignment does not happen. It still feels strange to have a purposefully unaligned field.

Reply:
All these structs have way more members than we use. I just chose to squash them all into _padN members instead of spelling them all out. I can drop a comment for this. (Or I am just not quite getting your meaning here :D)

(This one would actually normally be a uint64_t timestamp though.)

Reply:
OK, ignore my comment then.